src/zjs_modules.c

```diff
@@ -255,6 +266,53 @@ static ZJS_DECL_FUNC(process_exit)
     exit(status);
 }
 #endif
+#ifdef ZJS_DYNAMIC_LOAD
+void zjs_modules_check_load_file(char *file)
+{
+    // No file waiting to load, just return
+    if (load_file == NULL) {
+        return;
+    }
+
+    zjs_stop_js();
+    char *buf = NULL;
+    size_t size;
+    jerry_value_t parsed_code = 0;
+    buf = read_file_alloc(load_file, &size);
+    parsed_code = jerry_parse((const jerry_char_t *)buf, size, false);
+
+    if (jerry_value_has_error_flag(parsed_code)) {
+        ERR_PRINT("Error parsing JS\n");
+    }
```

Comment:
It looks like you're going to call `jerry_run` on this even though it was an error? Maybe change this to
```c
if (not_an_error) {
    keep_going();
}
else {
    complain();
}
free_stuff_up();
```

Reply:
Guess I was getting it confused with ZJS_ERROR - which would return. I'll add the check thanks.